src/components/Cell/Cell.tsx

```diff
@@ -35,26 +41,94 @@ const Cell = ({ id, bullDoze, setBullDoze, filterGarden, setFilterGarden, toggle
     setApiError(error)
   }
 
+  useEffect(() => {
+    if (garden) {
+      const foundCell = garden.find(cell => cell.id === id)
+      if (foundCell !== undefined && foundCell.status === 1) {
+        setIsDisabled(true);
+        !isDisabled ? setClassName('cell disabled') : setClassName('cell');
+      }
+      if (foundCell !== undefined && foundCell.plant_id) {
+        setCellContents({ plant: foundCell })
+        setIsPopulated(true);
+        setIsPlanted(true);
+      }
+      setCell(foundCell);
+    }
+    // eslint-disable-next-line
+  }, []);
+
+  useEffect(() => {
+    setGarden((prevState: GardenKeys) => {
+      let index = prevState?.findIndex((item) => item.id === cell?.id);
+      let newState = [...prevState];
+      newState[index] = {
+        ...newState[index],
+        status: Number(isDisabled)
+      };
+      return newState;
+    });
+    // eslint-disable-next-line
+  }, [isDisabled]);
+
+  useEffect(() => {
+    setGarden((prevState: GardenKeys) => {
+      let index = prevState?.findIndex((item) => item.id === cell?.id);
+      let newState = [...prevState];
+      newState[index] = {
+        ...newState[index],
+        image: cellContents?.plant.image,
+        name: cellContents?.plant.name,
+        'plant_id': cellContents?.plant.plant_id
+      };
+      return newState;
+    });
+    // eslint-disable-next-line
+  }, [isPlanted]);
+
   useEffect(() => {
     if (bullDoze) {
       setClassName('cell');
       emptyCell();
       setBullDoze(false);
+      setGarden((prevState: GardenKeys) => {
+        let index = prevState?.findIndex((item) => item.id === cell?.id);
+        let newState = [...prevState];
+        newState[index] = {
+          ...newState[index],
+          image: undefined,
+          name: undefined,
+          'plant_id': undefined,
+          status: 0
+        };
+        return newState;
+      });
```

Comment:
It's likely all we need to do for this one is trigger a re-render on the useEffect we run on page load.  Check it out when you're hooking this functionality into the BE.